mayastor/src/bdev/nexus/nexus_label.rs

```diff
@@ -367,61 +389,70 @@ pub struct GptHeader {
 
 impl GptHeader {
     pub const PARTITION_TABLE_SIZE: u64 = 128 * 128;
+    pub const DATA_OFFSET: u64 = 1024 * 1024;
```

Comment:
how is this offset and size calculated?

Reply:
The GPT spec definitely does not require this, but most partitioning tools will align partitions to 1 MiB boundaries. It is certainly the case that fdisk does this, and I suspect that is the main reason why we do also.
This means that the start of the first partition is always at an offset of 1024 * 1024 bytes (note this is regardless of the underlying block size).

Reply:
Also, the partition table size of 128 x 128 represents the "maximum" possible partition table size, although it could conceivably be smaller. However, we always reserve the maximum amount of space so that we can add partitions to the partition table without having to move existing partition boundaries. Again this is entirely consistent with how fdisk behaves.

Reply:
:+1: